bin/dump_kubernetes.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+#
+# Uses kubectl to collect cluster information.
+# Dumps:
+# - Logs of every container of every pod of every namespace.
+# - Resource configurations for ingress, endpoints, custom resource
+#   definitions, configmaps, secrets (names only) and "all" as defined by
+#   kubectl.
+
+OUT_DIR="${1:-istio-dump}"
+LOG_DIR="${OUT_DIR}/logs"
+RESOURCES_FILE="${OUT_DIR}/resources.yaml"
+
+check_prerequisites() {
+  PREREQUISITES=$*
+  for prerequisite in ${PREREQUISITES}; do
+    if ! command -v "${prerequisite}" > /dev/null; then
+      echo "\"${prerequisite}\" is required. Please install it."
+      return 1
+    fi
+  done
+}
+
+dump_logs() {
+  NAMESPACES=$(kubectl get namespaces -o=jsonpath="{.items[*].metadata.name}")
```

Comment:
Any reason `kubectl cluster-info dump --all-namespaces --output-directory=${OUT_DIR}` wouldn't work here?

Reply:
Mandar and I looked at that option, but it doesn't output a container's previous logs and is missing endpoints, configmaps, and secrets.